server/container/guice/guice-common/src/main/java/org/apache/james/PeriodicalHealthChecks.java

```diff
@@ -53,10 +57,38 @@ public void start() {
             .flatMap(any -> Flux.fromIterable(healthChecks)
                 .flatMap(healthCheck ->
                     Mono.fromCallable(healthCheck::check)))
+                .flatMap(result ->
+                    Mono.fromRunnable(() -> logResult(result)))
+            .onErrorContinue(this::logError)
             .subscribeOn(Schedulers.elastic())
             .subscribe();
     }
 
+    private void logResult(Result result) {
+        switch (result.getStatus()) {
+            case HEALTHY:
+                break;
+            case DEGRADED:
+                LOGGER.warn("DEGRADED: {} : {}", result.getComponentName().getName(), result.getCause());
+                break;
+            case UNHEALTHY:
+                if (result.getError().isPresent()) {
+                    LOGGER.error("UNHEALTHY: {} : {} : {}", result.getComponentName().getName(), result.getCause(), result.getError().get());
+                    break;
+                }
+
+                LOGGER.error("UNHEALTHY: {} : {}", result.getComponentName().getName(), result.getCause());
+                break;
+        }
+    }
+
+    private void logError(Throwable error, Object triggeringValue) {
+        if (triggeringValue instanceof Result) {
```

Comment:
any reason triggeringValue would be something else? What would that mean? Why would we log nothing?

Reply:
In this case we only have 1 Flux emitting Result elements so i think triggeringValue wouldn't be anything else. 

However the **onErrorContinue** operator accepts **Object** type since it can be applied to the whole chain of operation with different type may be? I haven't figured out a more elegant way to handle this.

Reply:
as it's unexpected we could at least log when there's something else